app/models/order.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class Order < ApplicationRecord
+  before_validation :generate_order_number, on: :create
+
+  belongs_to :user
+  has_many :order_items, dependent: :destroy
+
+  enum status: { unpaid: 'unpaid', paid: 'paid', expired: 'expired' }, _prefix: true
+
+  validates :number,
+            :currency,
+            :item_quantity,
+            :items_total,
+            :items_total_net,
+            :tax_total,
+            :discount_total,
+            :status,
+            presence: true
+  validates :number, uniqueness: true
```

Comment:
What will happen if generated number already exists in database?

Reply:
Currently the `generate_order_number` method checks if the given number exists. It may not be very beautiful, but it works 😄 If the number repeats somehow, rails will throw an exception.